app/models/concerns/phone_number_validator.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+# Validates that a given phone number (attribute) is valid
+class PhoneNumberValidator < ActiveModel::EachValidator
+  include ValidationHelper
+  def validate_each(record, attribute, value)
+    return if value.blank?
+
+    phone = Phonelib.parse(value, 'US')
```

Comment:
Yeah my hesitation in using that validation was because I was concerned our validation in `import_controller.rb` was different from just using `Phonelib.valid?` because maybe we wanted different behavior than what that function provides. But now that I've given it another look, I think our check of the length of the number is doing the same thing as `Phonelib.possible?`.  The docs on https://github.com/daddyz/phonelib#basic-usage refer me to https://github.com/google/libphonenumber#highlights-of-functionality, and the libphonenumber library indicates that `possible` checks based on length alone.

Reply:
So yes thanks for pointing this out, I will simplify this using the built in `Phonelib` method. I may still need to do a custom validator to construct the error message we want, but I can use `Phonelib.possible?` within that validator.

Reply:
If `.valid?` does more validation that we should be doing, I'd be okay using that as well. Looks like Tim wrote that line originally in `import_controller.rb`, might be worth asking him if there was any reason in particular he didn't use `.valid?` at that time.

Reply:
Talked to @timwongj about this, and he pointed out that we use a similar sort of logic on the [export side](https://github.com/SaraAlert/SaraAlert/blob/18dbc6855d8e5b7d48465d5a797ef0173b3133b1/app/lib/import_export.rb#L529) to format the phone number, so we'd need to be careful if we replace this. Tim also indicated he felt like he remembered some cases where `.valid?` would be `true` for some cases where we wanted `false`, but we haven't been able to recreate this testing right now. However, we have found that `.valid?` is definitely more strict, a number like `+15555550174` fails `.valid?` but passes our validation tests. It looks like the `.valid?` function actually checks if the number comes from a valid range of phone numbers, according to [this](https://github.com/google/libphonenumber/blob/master/FAQ.md#what-does-it-mean-for-a-phone-number-to-be-valid). So there are two things to consider there
1. Do we want that extra strictness at the price of some performance?
2. If we add this extra strictness, we will need to be careful about existing phone numbers. For example, the `+15555550174` number is from the example Excel spreadsheet that you can download as a part of the formatting guidance on sara alert.

Reply:
I don't think it's for me to say whether we should change the existing phone validation for the application, but regardless I don't think that's a change that should be made here as a part of this task. Just refactoring to use `possible?` is fine, since it sounds like that doesn't actually change the rule.

Reply:
I did more testing and I'm finding a weird discrepancy between `possible?` and our current approach. The `possible?` approach says that `123-456-7899` is an invalid phone number. Our existing approach sees it as valid. I think the possible approach is actually correct. Fun fact that I've just learned is that North American area codes I guess cannot start with `1`: https://en.wikipedia.org/wiki/North_American_Numbering_Plan. However, in the enrollment, you are able to make a new monitoree with a phone number that starts with `1`. I'm nervous about making the model validation out of sync with enrollment. You could end up with someone enrolling a monitoree and clicking "Finish" and then getting an error that the phone number is invalid, and I think it would make more sense to tell them that when they enter the phone number.

Reply:
Ok, I think we should just call this off then and just stick with the phone validation as is, as long as @kylietmo is ok with that. I created a placeholder ticket in the backlog to address improving phone validation across the app https://tracker.codev.mitre.org/browse/SARAALERT-969.

Reply:
Thank you Tom that's exactly what I was going to say we do. And thanks Nick for investigating this so thoroughly, completed agreed we should keep it consistent. Will resolve this thread.